docs/articles/viewing-epoch-based-metrics.rst

```diff
@@ -0,0 +1,207 @@
+   repository
+   <https://github.com/determined-ai/determined/tree/master/examples/tutorials/core_api_pytorch_mnist>`_.
+
+For this step, we'll use our ``model_def.py`` script and its accompanying ``const.yaml`` experiment
+configuration file.
+
+From the directory containing our files, we'll begin by running the following command:
+
+.. code:: bash
+
+   det e create const.yaml . -f
+
+We don't have any data to plot yet, but we'll visit the Determined WebUI to see that our experiment
+is running.
+
+To do this, we'll navigate to ``http://localhost:8080/``.
+
+To sign in, we'll accept the default determined username, leave the password empty, and click **Sign
+In**.
+
+In the WebUI, we'll select our experiment, and then navigate to the **Logs** tab.
+
+************************
+ Step 2: Report Metrics
+************************
+
+In this section, we'll define our epoch metric. We'll also report training and validation metrics to
+the Determined master. To do this, we'll import Determined, and create a
+:class:`~determined.core.Context` object to allow interaction with the master. Then, we'll pass the
+``core_context`` as an argument into ``main()``, ``train()``, and ``test()`` and modify the function
+headers accordingly.
+
+To follow along, use the ``model_def_metrics.py`` script and its accompanying ``metrics.yaml``
+experiment configuration file.
+
+We'll start by importing Determined:
+
+.. code:: bash
+
+   python3
+
+.. code:: python
+
+   import determined as det
+
+Step 2.1: Verify the Main Loop Contains ``core.Context``
+========================================================
+
+We'll need a ``core.Context`` object for interacting with the master. Verify the __main__loop in
+your script, ``model_def_metrics.py``, contains ``core_context``:
+
+.. code:: python
+
+   if __name__ == "__main__":
+
+     # NEW: Establish new determined.core.Context and pass to main
+     # function.
+     with det.core.init() as core_context:
+         main(core_context=core_context)
+
+Step 2.2: Modify the Train and Validation Methods to Report Epoch-Based Metrics
+===============================================================================
+
+Our script already contains ``core_context.train``. This is used to report training and validation
+metrics.
+
+But, we also want to report epoch-based metrics. To do this, we'll modify the train() method to
+include ``epoch_idx`` as a metric. This allows Determined to keep track of the specific epoch for
+which training loss is being reported:
+
+.. code:: python
+
+   # NEW: Report epoch-based training metrics to Determined
+   # master via core_context.
+   # Index by (batch_idx + 1) * (epoch-1) * len(train_loader)
+   # to continuously plot loss on one graph for consecutive
+   # epochs.
+   core_context.train.report_training_metrics(
+       steps_completed=batches_completed + epoch_idx * len(train_loader),
+       metrics={"train_loss": loss.item(), "epoch": epoch_idx},
```

Comment:
Is all you need to do to get epoch indices to pass `"epoch": epoch_idx`?

I feel like this is several pages of docs to explain just that.

Is there somewhere we can say that in a much, much more concise way?

Reply:
per our discussion, this is resolved